src/Database/Type/EnumType.php

```diff
@@ -0,0 +1,97 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         5.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Database\Type;
+
+use BackedEnum;
+use Cake\Database\Driver;
+use InvalidArgumentException;
+use PDO;
+
+/**
+ * Enum type converter.
+ *
+ * Use to convert string data between PHP and the database types.
+ */
+class EnumType extends BaseType
+{
+    /**
+     * Convert enum instances into the database format.
+     *
+     * @param mixed $value The value to convert.
+     * @param \Cake\Database\Driver $driver The driver instance to convert with.
+     * @return string|int|null
+     */
+    public function toDatabase(mixed $value, Driver $driver): string|int|null
+    {
+        if ($value === null) {
+            return null;
+        }
+
+        if ($value instanceof BackedEnum) {
+            return $value->value;
+        }
+
+        if (is_int($value) || is_string($value)) {
```

Comment:
Shouldn't we require the value is in BackedEnum form?

Reply:
The fact that users can still patch entities with scalar enum representations via `->patchEntity()` is good enough for me. They shouldn't directly set the fields and save the entity with scalar enum values.

Reply:
I don't think we should. Foe e.g. for a datetime field we can directly set a string and are not forced to use an object.

Reply:
> They shouldn't directly set the fields and save the entity with scalar enum values.

That would be inconsistent with how other types which are represented as objects in PHP behave.

Reply:
But strings are always valid for datetime fields in the db. 

The enum type should really be instanced with the enum class it expects as input and should resolve to. However, we can't do that with the type system so we have this Behavior that is expected to marshal the data. The type is useless without the behavior unfortunately.

However, the check for BackedEnum is also not useful since it doesn't mean anything without the Behavior.

*shrug*

Reply:
In that case I am leaning towards having a consistent behavior 😄

Reply:
> But strings are always valid for datetime fields in the db.

It's the same for enum fields. Enums values are strings in the query/db.

Regardless of the limitation of the EnumType and it's reliance on the behavior one should still be able to directly set values which are accepted by the db, same as you can for all other types which are represented as objects in PHP.

Reply:
Oh, I thought it was supporting all types that BackedEnum does. In that case, never mind.